test/extensions/filters/http/oauth2/oauth_integration_test.cc

```diff
@@ -480,8 +481,33 @@ TEST_P(OauthIntegrationTest, RefreshTokenFlow) {
   doRefreshTokenFlow("token_secret_1", "hmac_secret_1");
 }
 
-// Regression test(issue #22678) where (incorrectly)using server's init manager(initialized state)
-// to add init target by the secret manager led to the assertion failure.
+// Verify the behavior when the callback param is missing the state query param.
+TEST_P(OauthIntegrationTest, MissingStateParam) {
```

Comment:
I moved this up from below so it's grouped with the other `OauthIntegrationTest`s instead of `OauthIntegrationTestWithBasicAuth`.